### PR TITLE
Clean up `Modal::class`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Modal.php [hunk 1]
```diff
@@ -29,176 +31,251 @@
...
     private string $toggleButtonLabel = 'Toggle button';
     private string $toggleButtonSize = '';
     private string $toggleButtonColor = '';
-    private array $toggleButtonOptions = [];
+    private bool $withoutCloseButton = true;
     private bool $withoutToggleButton = true;
 
     /**
-     * Returns a new instance with the specified main container options.
+     * The HTML attributes. The following special options are recognized.
```
Comment (samdark) [inline src/Modal.php:54]: There are no special options info so

```suggestion
     * The HTML attributes.
```

File: src/Modal.php [hunk 2]
```diff
@@ -29,176 +31,251 @@
...
 
         return $new;
     }
 
     /**
-     * Returns a new instance with the specified toggle button label.
+     * Returns a new instance with the specified close button size.
      *
-     * @param string $value The toggle button label.
+     * @param string $value The close button size.
```
Comment (samdark) [inline src/Modal.php:103]: Should mention what values are accepted. Also, maybe we can remove `is-` prefix?

File: src/Modal.php [hunk 3]
```diff
@@ -29,176 +31,251 @@
...
+        $new = clone $this;
+        $new->toggleButtonAttributes = $values;
+
+        return $new;
+    }
+
+    /**
+     * Returns a new instance with the specified toggle button color.
+     *
+     * @param string $value The toggle button color.
```
Comment (samdark) [inline src/Modal.php:231]: Accepted values should be listed here. Also, maybe we can remove `is-` prefix?

File: src/Modal.php [hunk 4]
```diff
@@ -29,176 +31,251 @@
...
+        $new = clone $this;
+        $new->toggleButtonLabel = $value;
+
+        return $new;
+    }
+
+    /**
+     * Returns a new instance with the specified toggle button size.
+     *
+     * @param string $value The toggle button size.
```
Comment (samdark) [inline src/Modal.php:266]: Accepted values should be listed here. Also, maybe we can remove `is-` prefix?

File: src/Modal.php [hunk 5]
```diff
@@ -217,18 +294,14 @@ public function withoutCloseButton(): self
...
-     *
-     * @param array $value The content options.
-     *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * Returns a new instance with the disabled toggle button.
      *
      * @return self
      */
-    public function contentOptions(array $value): self
+    public function withoutToggleButton(): self
```
Comment (samdark) [inline src/Modal.php:301]: Majority of options accept boolean value. Should we do it here as well?

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Modal.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Modal.php:54]: done.

File: src/Modal.php (hunk 5 above)
Comment (terabytesoftw) [reply src/Modal.php:301]: done.

File: src/Modal.php (hunk 4 above)
Comment (terabytesoftw) [reply src/Modal.php:266]: done.

File: src/Modal.php (hunk 3 above)
Comment (terabytesoftw) [reply src/Modal.php:231]: the prefix is part of the bulma classes.

File: src/Modal.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Modal.php:103]: done.

File: src/Modal.php (hunk 3 above)
Comment (terabytesoftw) [reply src/Modal.php]: done.

File: docs/modal.md [hunk 6]
```diff
@@ -25,48 +25,104 @@ use Yiisoft\Yii\Bulma\Asset\BulmaJsAsset;
...
 Method | Description | Default
 -------|-------------|---------
-`id(string $value)` | Widget ID. | `''`
+`attributes(array $value)` | The HTML attributes. | `[]`
 `autoIdPrefix(string $value)` | Prefix to the automatically generated widget ID. | `w`
-`options(array $value)` | HTML attributes for the widget container tag. | [`class` => `modal`]
-`contentOptions(array $value)`| HTML attributes for the widget content tag. | `[]`
-`closeButtonOptions(array $value)` | HTML attributes for the widget button tag. | [`class` => `modal-close`, `aria-label` => `close`]
-`closeButtonSize(string $value)` | Size close button. | `is-small`, `is-medium`, `is-large`
+`closeButtonAttributes(array $value)` | HTML attributes for the button tag. | [`class` => `modal-close`, `aria-label` => `close`]
```
Comment (samdark) [inline docs/modal.md:116]:
```suggestion
`closeButtonAttributes(array $value)` | HTML attributes for the close button tag. | [`class` => `modal-close`, `aria-label` => `close`]
```

File: docs/modal.md [hunk 7]
```diff
@@ -25,48 +25,104 @@ use Yiisoft\Yii\Bulma\Asset\BulmaJsAsset;
...
+`closeButtonSize(string $value)` | Size close button. Options available are: (`Modal::SIZE_SMALL`, `Modal::SIZE_MEDIUM`, `Modal::SIZE_LARGE`). | Default setting empty normal.
```
Comment (samdark) [inline docs/modal.md:117]:
```suggestion
`closeButtonSize(string $value)` | Close button size. Options available are: (`Modal::SIZE_SMALL`, `Modal::SIZE_MEDIUM`, `Modal::SIZE_LARGE`). | Default size is normal.
```

File: docs/modal.md [hunk 8]
```diff
@@ -25,48 +25,104 @@ use Yiisoft\Yii\Bulma\Asset\BulmaJsAsset;
...
+`contentAttributes(array $value)`| HTML attributes for the content tag. | `[]`
+`id(string $value)` | Widget ID. | `''`
+`modalBackgroundClass(string $value)` | Class for the modal background. | `modal-background`
```
Comment (samdark) [inline docs/modal.md:120]: I don't think we need to mention "modal".

File: src/Modal.php [hunk 9]
```diff
@@ -44,191 +46,293 @@ final class Modal extends Widget
...
+    }
 
+    /**
+     * Returns a new instance with the specified modal background class.
+     *
+     * @param string $value The modal background class.
+     *
+     * @return self
+     */
+    public function modalBackgroundClass(string $value): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Modal.php]: done.

File: src/Modal.php [hunk 10]
```diff
@@ -44,191 +46,293 @@ final class Modal extends Widget
...
-     * Returns a new instance with the specified toggle button color.
+     * Returns a new instance with the specified modal button class.
      *
-     * @param string $value The toggle button color.
+     * @param string $value The modal button class.
      *
      * @return self
      */
-    public function toggleButtonColor(string $value): self
+    public function modalButtonClass(string $value): self
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/Modal.php]: done.

File: docs/modal.md (hunk 6 above)
Comment (terabytesoftw) [reply docs/modal.md:118]: done.

File: docs/modal.md (hunk 7 above)
Comment (terabytesoftw) [reply docs/modal.md:119]: done.

File: docs/modal.md (hunk 8 above)
Comment (terabytesoftw) [reply docs/modal.md]: done.

File: src/Modal.php [hunk 11]
```diff
@@ -44,191 +46,293 @@ final class Modal extends Widget
...
+     * @param string $value The close button size. Default setting empty normal.
+     * Possible values: Modal::SIZE_SMALL, Modal::SIZE_MEDIUM, Model::SIZE_LARGE.
      *
      * @return self
      */
-    public function toggleButtonLabel(string $value): self
+    public function closeButtonSize(string $value): self
     {
+        if (!in_array($value, self::SIZE_ALL, true)) {
+            $values = implode('"', self::SIZE_ALL);
```
Comment (samdark) [inline src/Modal.php:134]:
```suggestion
            $values = implode('", "', self::SIZE_ALL);
```

File: src/Modal.php [hunk 12]
```diff
@@ -44,191 +46,293 @@ final class Modal extends Widget
...
+     * @param string $value The toggle button color. Default setting is empty without color.
+     * Possible values: Modal::COLOR_PRIMARY, Modal::COLOR_LINK, Modal::COLOR_INFO, Modal::COLOR_SUCCESS,
+     * Modal::COLOR_WARNING, Modal::COLOR_DANGER, Modal::COLOR_DARK.
+     *
+     * @return self
+     */
+    public function toggleButtonColor(string $value): self
+    {
+        if (!in_array($value, self::COLOR_ALL, true)) {
+            $values = implode(' ', self::COLOR_ALL);
```
Comment (samdark) [inline src/Modal.php:264]:
```suggestion
            $values = implode('", "', self::COLOR_ALL);
```

File: src/Modal.php [hunk 13]
```diff
@@ -44,191 +46,293 @@ final class Modal extends Widget
...
+     * @param string $value The toggle button size. Default setting empty normal.
+     * Possible values: Modal::SIZE_SMALL, Modal::SIZE_MEDIUM, Model::SIZE_LARGE.
      *
      * @return self
      */
-    public function closeButtonOptions(array $value): self
+    public function toggleButtonSize(string $value): self
     {
+        if (!in_array($value, self::SIZE_ALL, true)) {
+            $values = implode(' ', self::SIZE_ALL);
```
Comment (samdark) [inline src/Modal.php:300]:
```suggestion
            $values = implode('", "', self::SIZE_ALL);
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Modal.php (hunk 11 above)
Comment (terabytesoftw) [reply src/Modal.php:134]: done.

File: src/Modal.php (hunk 12 above)
Comment (terabytesoftw) [reply src/Modal.php:264]: done.

File: src/Modal.php (hunk 13 above)
Comment (terabytesoftw) [reply src/Modal.php:300]: done.

**Pull request merged**